### DISCUSSION TITLE
发布响应模式的使用

### DISCUSSION THREAD

**Discussion by githubzhangfei** [nats-io/nats-server#5165] [Q&A] [answered]
我能不能用这个模式在某些场景下取代http的请求响应？我尝试了一下，回复的消息只有一个主题。并没有找到回复的内容。我的需求如下：第一步：我向主题a发送消息。订阅者b收到后，根据收到的结果进行逻辑处理。然后把结果返回给a。我应该怎么样才能拿到结果呢？[测试代码1](https://github.com/nats-io/nats.java/blob/main/src/examples/java/io/nats/examples/NatsReqFuture.java)[测试代码2](https://github.com/nats-io/nats.java/blob/main/src/examples/java/io/nats/examples/NatsReply.java)从上面代码中摘录的api。虚心求教

**Comment by derekcollison** [comment, marked as answer]
Yes you can use this for that purpose. Here is an example.

https://github.com/derekcollison/nats-http
